src/PantheonTerminalWindow.vala

```diff
@@ -306,19 +306,25 @@ namespace PantheonTerminal {
                             return true;
                         }
                     } else if (match_keycode (Gdk.Key.v, keycode)) {
-                        if (search_toolbar.search_entry.has_focus) {
-                            return false;
-                        } else if (clipboard.wait_is_text_available ()) {
-                            action_paste ();
-                            return true;
-                        }
+                        handle_paste_event ();
                     }
                 }
 
                 return false;
             });
         }
 
+        public bool handle_paste_event () {
+            if (this.search_toolbar.search_entry.has_focus) {
```

Comment:
Unnecessary use of ****this****.  There is no ambiguity and rest of this class does not use this form for `searchbar`.